src/halmos/sevm.py

```diff
@@ -585,6 +585,53 @@ def __next__(self) -> Instruction:
         return insn
 
 
+class Path:
+    solver: Solver
+    num_scopes: int
+    conditions: List
+    pending: List
+
+    def __init__(self, solver: Solver):
+        self.solver = solver
+        self.num_scopes = 0
+        self.conditions = []
+        self.pending = []
+
+    def __deepcopy__(self, memo):
+        if len(self.pending) > 0:
+            raise ValueError("deepcopy pending path", self)
+
+        path = Path(self.solver)
```

Comment:
might be worth a comment here. This is where the real magic happens since we re-use the solver instance instead of actually copying it, right? Maybe explain in a sentence or two why we do this and the trade-offs